api/coreservice.go

```diff
@@ -1929,20 +1983,42 @@ func (core *coreService) traceTx(ctx context.Context, txctx *tracers.Context, co
 	return retval, receipt, tracer, err
 }
 
-func (core *coreService) simulateExecution(ctx context.Context, addr address.Address, elp action.Envelope, opts ...protocol.SimulateOption) ([]byte, *action.Receipt, error) {
-	ctx, err := core.bc.Context(ctx)
+func (core *coreService) simulateExecution(
+	ctx context.Context,
+	height uint64,
+	archive bool,
```

Comment:
can consider passing 'ws' as a parameter instead of the boolean.

Reply:
that's doable, yet would cause code L2004-2008 to be duplicated at these funcs:
```
func (core *coreService) ReadContract()
func (core *coreService) isGasLimitEnough()
func (core *coreService) SimulateExecution()
func (core *coreService) TraceTransaction()
func (core *coreService) TraceCall()
```
IMO the current way is more efficient